src/test_application/main.cpp

```diff
@@ -317,26 +320,28 @@ void debug_print_hier()
 
 void debug_print_sats()
 {
+    using osp::universe::UCompTransformTraj;
+    using osp::universe::UCompType;
 
-    universe::Universe &universe = g_osp.get_universe();
+    osp::universe::Universe const &universe = g_osp.get_universe();
 
-    auto view = universe.get_reg().view<universe::UCompTransformTraj,
-                                        universe::UCompType>();
+    auto const view = universe.get_reg().view<const UCompTransformTraj,
+                                              const UCompType>();
 
-    for (universe::Satellite sat : view)
+    for (osp::universe::Satellite sat : view)
```

Comment:
Are osp::Universe::Satellite's cheap to copy-construct? That's what the for-loop is doing for every item in the view.

If not cheap, then recommend const&

Reply:
The "view" object in these situations is a collection of entities that meet a certain criterion based on the components they have, but the iteration is over the entities themselves. The entities themselves (e.g. `Satellite`) are just an ID represented by a `uint32_t` so there's no performance concern when dealing with them.